### PR TITLE
Report the effective dedicated server mode to serverlist

### PR DESCRIPTION
**Pull request by kwolekr**
This fixes the case where servers started from within the game do not always have their dedicated server status reported correctly.

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -148,8 +148,8 @@ Server::Server(
 		const std::string &path_world,
 		const SubgameSpec &gamespec,
 		bool simple_singleplayer_mode,
-		bool ipv6
-	):
+		bool ipv6,
```
Comment (kwolekr) [inline src/server.cpp]: Hmmm, instead of continuously adding more boolean parameters, why not combine these into a set of "Server Startup" flags?

Comment (est31) [inline src/server.cpp]: Perhaps the compiler already does that, doesn't it?
Better we should have a complete struct with other stuff than just bools, like the MainMenuData struct.

**Comment by kwolekr** [comment]
I am not aware of any compiler optimization which does this.
The main reason for changing many bool parameters to flags imho should not be for performance, but instead expandability/clarity.  If a new boolean value gets added, you don't need to modify the interface of the class at all.  More important is that what the boolean actually is gets repeated in the parameter list so it's instantly clear without having to look at the class definition.  i.e. DoThing(false, false, true, false) vs. DoThing(THING_OPTION_BUFFERED | THING_OPTION_DISABLE_OTHER_THING, ...);
Which would you say is more clear?

**Comment by est31** [comment]
You gain clarity at the caller part, but you lose clarity at the part where the method is declared and defined.

**Comment by est31** [comment]
Also, the type system doesn't protect you from anything anymore once you send an int like type, instead of bools. You could send some completely unrelated flags and the compiler wouldn't notice.

**Comment by est31** [comment]
https://github.com/minetest/minetest/pull/2604/files#diff-18513665750ef5adf42b5ec29e14162eR1453

Why no reference?

**Comment by kwolekr** [comment]
All valid points.  I suppose it comes down to preference.

**Comment by ShadowNinja** [comment]
b8484ef24e8e1ec90a2967372808ab2bea538c7c

**Pull request closed without merging**